lib/api/src/sys/function_env.rs

```diff
@@ -40,16 +40,18 @@ impl<T> FunctionEnv<T> {
             .unwrap()
     }
 
-    /// Get the data as mutable
-    pub fn as_mut<'a>(&self, store: &'a mut impl AsStoreMut) -> &'a mut T
+    /// Get the data as mutable reference
+    /// (this will only return a mutable reference as long as the environment
+    ///  has not been cloned - environments are cloned during multithreading)
+    pub fn as_mut<'a>(&mut self, store: &'a mut impl AsStoreMut) -> Option<&'a mut T>
     where
         T: Any + Send + 'static + Sized,
     {
         self.handle
             .get_mut(store.objects_mut())
             .as_mut()
-            .downcast_mut::<T>()
-            .unwrap()
+            .map(|a| a.downcast_mut::<T>())
+            .flatten()
```

Comment:
I think `.unwrap().unwrap()` will be faster to execute?

Reply:
It probably compiles to the same assembly code - I can change it if you like though no worries